spec/queryable_spec.cr

```diff
@@ -765,14 +765,13 @@ describe Avram::Queryable do
       query.select_count.should eq 1
     end
 
-    it "raises when used with offset or limit" do
-      expect_raises(Avram::UnsupportedQueryError) do
-        UserQuery.new.limit(1).select_count
-      end
```

Comment:
You can now use a limit and offset in the select_count queries. The reason you couldn't before was because `SELECT COUNT()` would always ever return 1 row. So it didn't make sense to have a limit or offset. Now with this sub-select way, you can use limit to your advantage.

Say you have a table with 7 records in it. If you do a limit 10, and then run a count, you'll get 7. Do the same thing once there's 12 records in the table, and then you'll only get 10. It's sort of like an upper bounds check.